routing/world_graph.cpp

```diff
@@ -4,12 +4,45 @@ namespace routing
 {
 using namespace std;
 
+void WorldGraph::GetTwins(Segment const & segment, bool isOutgoing, vector<SegmentEdge> & edges)
+{
+  vector<Segment> twins;
+  GetTwinsImpl(segment, isOutgoing, twins);
+
+  if (GetMode() == Mode::LeapsOnly)
+  {
+    // Ingoing edges listing is not supported in LeapsOnly mode because we do not have enough
+    // information to calculate |segment| weight. See https://jira.mail.ru/browse/MAPSME-5743 for details.
+    CHECK(isOutgoing, ("Ingoing edges listing is not supported in LeapsOnly mode."));
+    // We need both enter to mwm and exit from mwm in LeapsOnly mode to reconstruct leap.
+    // That's why we need to duplicate twin segment here and than remove duplicate
+    // while processing leaps.
+    for (Segment const & twin : twins)
+    {
+      m2::PointD const & from = GetPoint(segment, isOutgoing /* front */);
+      m2::PointD const & to = GetPoint(twin, isOutgoing /* front */);
+      // Weight is usually zero because twins correspond the same feature
+      // in different mwms. But if we have mwms with different versions and feature
+      // was moved in one of them we can have nonzero weight here.
```

Comment:
`we can have nonzero weight here.`->`the weight is greater than zero.`

Reply:
done